src/components/activity-page/managed-interactive/managed-interactive.tsx

```diff
@@ -16,86 +17,122 @@ interface IProps {
-    let embeddableData: IMwInteractive | LibraryInteractiveData;
-    if (embeddable.type === "ManagedInteractive") {
-      embeddableData = embeddable.library_interactive.data;
-    } else {
-      embeddableData = embeddable;
+  const [showModal, setShowModal] = useState(false);
+
+  // both Modal and inline versions of interactive should reflect the same state
+  const iframeInteractiveState = useRef(embeddable.interactiveState);
+  const handleNewInteractiveState = (state: any) => {
+    iframeInteractiveState.current = state;
+    // TODO: handle interactive state storage
+  };
+
+  const questionName = embeddable.name ? `: ${embeddable.name}` : "";
+  // in older iframe interactive embeddables, we get url, native_width, native_height, etc. directly off
+  // of the embeddable object. On newer managed/library interactives, this data is in library_interactive.data.
+  let embeddableData: IMwInteractive | LibraryInteractiveData;
+  if (embeddable.type === "ManagedInteractive") {
+    embeddableData = embeddable.library_interactive.data;
+  } else {
+    embeddableData = embeddable;
+  }
+  const url = embeddableData.base_url || embeddableData.url || "";
+  // TODO: handle different aspect ration methods
+  // const aspectRatioMethod = data.aspect_ratio_method ? data.aspect_ratio_method : "";
+  const nativeHeight = embeddableData.native_height || 0;
+  const nativeWidth = embeddableData.native_width || 0;
+  const aspectRatio = nativeHeight && nativeWidth ? nativeWidth / nativeHeight : kDefaultAspectRatio;
+
+  // cf. https://www.npmjs.com/package/@react-hook/resize-observer
+  const useSize = (target: any) => {
+    const [size, setSize] = React.useState();
+
+    React.useLayoutEffect(() => {
+      setSize(target.current.getBoundingClientRect());
+    }, [target]);
+
+    useResizeObserver(target, (entry: any) => setSize(entry.contentRect));
+    return size;
+  };
+
+  const divTarget = React.useRef(null);
+  const divSize: any = useSize(divTarget);
+  const proposedHeight: number = divSize && divSize.width / aspectRatio;
+  const containerWidth: number = divSize && divSize.width;
+
+  const embeddableAuthoredState = () => {
+    const parsedState = embeddable.authored_state ? JSON.parse(embeddable.authored_state) : {};
+    parsedState.modalSupported = true;
+    return JSON.stringify(parsedState);
+  };
+
+  const [showHint, setShowHint] = useState(false);
+  const [hint, setHint] = useState("");
+  const handleHintCloseClick = () => {
+    setShowHint(false);
+  };
+
+  const handleQuestionClick = () => {
+    setShowHint(!showHint);
+  };
+
+  const setNewHint = useCallback((newHint: string) => {
+    setHint(newHint);
+  }, []);
+
+  const getModalContainer = () => {
+    const modalParent = document.getElementById("app");
+    const container = modalParent ? modalParent : {};
+    return container;
+  };
+
+  const toggleModal = () => {
+    setShowModal(!showModal);
+  };
+
+  const closeModal = (modalProps?: any) => {
```

Comment:
```suggestion
  const handleCloseModalRequest = (modalProps?: any) => {
```
`closeModal` sounds like an imperative, particularly next to the imperative `toggleModal()` above.